nmrfx-processor/src/main/java/org/nmrfx/processor/datasets/vendor/JCAMPData.java

```diff
@@ -17,987 +17,569 @@
-            d = 298.0;
+    public double[] getCoefs(int dim) {
+        //XXX Was not implemented in original JCAMPData.
+        //Inspired from BrukerData instead.
+        if (dim == 0) {
+            return new double[0];
         }
-        return d;
-    }
 
-    @Override
-    public String getSequence() {
-        String s;
-        if ((s = getPar("PULPROG,1")) == null) {
-            s = "";
+        FnMode fnMode = getFnMode(dim);
+        if (fnMode == null || fnMode == FnMode.QF || fnMode == FnMode.QSEQ || fnMode == FnMode.TPPI) {
+            return new double[0];
+        } else if (fnMode == FnMode.STATES) {
+            return new double[]{1, 0, 0, 0, 0, 0, 1, 0};
+        } else if (fnMode == FnMode.UNDEFINED || fnMode == FnMode.STATES_TPPI) {
+            return new double[]{1, 0, 0, 0, 0, 0, 1, 0};
+        } else if (fnMode == FnMode.ECHO_ANTIECHO) {
+            return new double[]{1, 0, -1, 0, 0, 1, 0, 1};
         }
-        return s;
+        return new double[]{1, 0, 0, 1};
     }
 
     @Override
-    public long getDate() {
-        String s;
-        long seconds = 0;
-        if ((s = getPar("DATE,1")) != null) {
-            try {
-                System.out.println("sec " + s);
-                seconds = Long.parseLong(s);
-            } catch (NumberFormatException e) {
-            }
-        } else {
-            System.out.println("no date");
+    public String getSymbolicCoefs(int dim) {
+        //XXX Was not implemented in original JCAMPData.
+        //Inspired from BrukerData instead.
+        if (dim == 0) {
+            return null;
         }
-        return seconds;
-    }
 
-    // open and read Bruker text file
-    @Override
-    public String getText() {
-        if (text == null) {
-            String textPath = "";
-            text = "";
+        FnMode fnMode = getFnMode(dim);
+        if (fnMode == null) {
+            return null;
+        } else if (fnMode == FnMode.QSEQ || fnMode == FnMode.TPPI) {
+            return "real";
+        } else if (fnMode == FnMode.STATES) {
+            return "hyper-r";
+        } else if (fnMode == FnMode.UNDEFINED || fnMode == FnMode.STATES_TPPI) {
+            return "hyper";
+        } else if (fnMode == FnMode.ECHO_ANTIECHO) {
+            return "echo-antiecho-r";
         }
-        return text;
+        return "sep";
     }
 
     @Override
-    public double getPH0(int iDim) {
-        double ph0 = 0.0;
-        Double dpar;
-        if ((dpar = getParDouble("PHC0," + (iDim + 1))) != null) {
-            ph0 = -dpar;
-            if (iDim == 0) {
-                ph0 += 90.0;
-            } else if (iDim == 1) {
-                ph0 += deltaPh0_2;
-            }
-        }
-        return ph0;
+    public String getVendor() {
+        //XXX some code expected "bruker" in lowercase
+        // RefManager.setupItems() to add getNDim() in a string builder
+        // RefManager.setupItems() to add a getFixDSP() option
```

Comment:
Returning something other that "bruker" should be correct but will have some behavior change